main.tf

```diff
@@ -223,13 +223,13 @@ resource "ibm_cos_bucket" "cos_bucket1" {
 }
 
 locals {
-  bucket_crn           = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].crn : ibm_cos_bucket.cos_bucket1[*].crn
-  bucket_id            = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].id : ibm_cos_bucket.cos_bucket1[*].id
-  bucket_name          = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].bucket_name : ibm_cos_bucket.cos_bucket1[*].bucket_name
-  bucket_storage_class = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].storage_class : ibm_cos_bucket.cos_bucket1[*].storage_class
-  s3_endpoint_public   = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].s3_endpoint_public : ibm_cos_bucket.cos_bucket1[*].s3_endpoint_public
-  s3_endpoint_private  = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].s3_endpoint_private : ibm_cos_bucket.cos_bucket1[*].s3_endpoint_private
-  s3_endpoint_direct   = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].s3_endpoint_direct : ibm_cos_bucket.cos_bucket1[*].s3_endpoint_direct
+  bucket_crn           = var.create_cos_bucket ? (var.encryption_enabled == true ? tolist(ibm_cos_bucket.cos_bucket[*].crn)[0] : tolist(ibm_cos_bucket.cos_bucket1[*].crn)[0]) : null
+  bucket_id            = var.create_cos_bucket ? (var.encryption_enabled == true ? tolist(ibm_cos_bucket.cos_bucket[*].id)[0] : tolist(ibm_cos_bucket.cos_bucket1[*].id)[0]) : null
+  bucket_name          = var.create_cos_bucket ? (var.encryption_enabled == true ? tolist(ibm_cos_bucket.cos_bucket[*].bucket_name)[0] : tolist(ibm_cos_bucket.cos_bucket1[*].bucket_name)[0]) : null
+  bucket_storage_class = var.create_cos_bucket ? (var.encryption_enabled == true ? tolist(ibm_cos_bucket.cos_bucket[*].storage_class)[0] : tolist(ibm_cos_bucket.cos_bucket1[*].storage_class)[0]) : null
+  s3_endpoint_public   = var.create_cos_bucket ? (var.encryption_enabled == true ? tolist(ibm_cos_bucket.cos_bucket[*].s3_endpoint_public)[0] : tolist(ibm_cos_bucket.cos_bucket1[*].s3_endpoint_public)[0]) : null
+  s3_endpoint_private  = var.create_cos_bucket ? (var.encryption_enabled == true ? tolist(ibm_cos_bucket.cos_bucket[*].s3_endpoint_private)[0] : tolist(ibm_cos_bucket.cos_bucket1[*].s3_endpoint_private)[0]) : null
+  s3_endpoint_direct   = var.create_cos_bucket ? (var.encryption_enabled == true ? tolist(ibm_cos_bucket.cos_bucket[*].s3_endpoint_direct)[0] : tolist(ibm_cos_bucket.cos_bucket1[*].s3_endpoint_direct)[0]) : null
```

Comment:
Everywhere there is a tolist, it is known that there is exactly on cos_bucket. So is there any value of taking one property out of that definition in a list and taking the first item in that list?

Or is

    bucket_crn           = var.create_cos_bucket ? (var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[0].crn : ibm_cos_bucket.cos_bucket1[0].crn) : null
    bucket_id            = var.create_cos_bucket ? (var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[0].id : ibm_cos_bucket.cos_bucket1[0].id) : null
    bucket_name          = var.create_cos_bucket ? (var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[0].bucket_name : ibm_cos_bucket.cos_bucket1[0].bucket_name) : null
    bucket_storage_class = var.create_cos_bucket ? (var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[0].storage_class : ibm_cos_bucket.cos_bucket1[0].storage_class) : null
    s3_endpoint_public   = var.create_cos_bucket ? (var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[0].s3_endpoint_public : ibm_cos_bucket.cos_bucket1[0].s3_endpoint_public) : null
    s3_endpoint_private  = var.create_cos_bucket ? (var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[0].s3_endpoint_private : ibm_cos_bucket.cos_bucket1[0].s3_endpoint_private) : null
    s3_endpoint_direct   = var.create_cos_bucket ? (var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[0].s3_endpoint_direct : ibm_cos_bucket.cos_bucket1[0].s3_endpoint_direct) : null

simpler?

The same can be said at L59-60

    cos_instance_id          = var.create_cos_instance == true ? ibm_resource_instance.cos_instance[0].id : var.existing_cos_instance_id
    cos_instance_guid        = var.create_cos_instance == true ? ibm_resource_instance.cos_instance[0].guid : element(split(":", var.existing_cos_instance_id), length(split(":", var.existing_cos_instance_id)) - 3)

Reply:
Updated the values to include the actual index rather than wildcard then access